cluster/manifests/skipper/deployment.yaml

```diff
@@ -208,40 +220,50 @@ spec:
           readOnlyRootFilesystem: true
           runAsNonRoot: true
           runAsUser: 1000
-{{ if and (eq .ConfigItems.skipper_local_tokeninfo "bridge") (eq .ConfigItems.enable_apimonitoring "true")}}
-        volumeMounts:
-          - name: routes
-            mountPath: /etc/routes
-          - name: filters
-            mountPath: /etc/config/default-filters
-{{ else if eq .ConfigItems.skipper_local_tokeninfo "bridge"}}
+{{ if or (eq .Cluster.ConfigItems.skipper_local_tokeninfo "bridge") (eq .Cluster.ConfigItems.enable_apimonitoring "true") (eq .Cluster.ConfigItems.skipper_oauth2_ui_login "true")}}
         volumeMounts:
+{{ end }}
+{{ if eq .Cluster.ConfigItems.skipper_local_tokeninfo "bridge"}}
           - name: routes
             mountPath: /etc/routes
-{{ else if eq .ConfigItems.enable_apimonitoring "true"}}
-        volumeMounts:
+{{ end }}
+{{ if eq .Cluster.ConfigItems.enable_apimonitoring "true"}}
           - name: filters
             mountPath: /etc/config/default-filters
+{{ end }}
+{{ if eq .Cluster.ConfigItems.skipper_oauth2_ui_login "true"}}
+          - name: oauth2
+            mountPath: {{ .Cluster.ConfigItems.skipper_oauth2_secrets_dir }}
+            readOnly: true
+          - name: encryption-key
+            mountPath: /etc/skipper/secret
+            readOnly: true
 {{ end }}
       securityContext:
         fsGroup: 1000
-{{ if and (eq .ConfigItems.skipper_local_tokeninfo "bridge") (eq .ConfigItems.enable_apimonitoring "true")}}
+{{ if or (eq .Cluster.ConfigItems.skipper_local_tokeninfo "bridge") (eq .Cluster.ConfigItems.enable_apimonitoring "true") (eq .Cluster.ConfigItems.skipper_oauth2_ui_login "true")}}
```

Comment:
Do we really need this? `volumes:` without any contents works fine I think.

Reply:
I will try

Reply:
looks like it drops it if you send an empty list for volumes and volumeMounts

Reply:
fixed